gpflow/covariances/mo_kuus.py

```diff
@@ -2,71 +2,70 @@
 
 import tensorflow as tf
 
-from ..features import (InducingPoints, MixedKernelSharedMof,
-                        SeparateIndependentMof, SharedIndependentMof)
-from ..kernels import (Mok, SeparateIndependentMok, SeparateMixedMok,
-                       SharedIndependentMok)
+from ..inducing_variables import (InducingPoints, FallbackSharedIndependentInducingVariables,
+                                  FallbackSeparateIndependentInducingVariables)
+from ..kernels import (MultioutputKernel, SeparateIndependent, LinearCoregionalisation,
+                       SharedIndependent)
 from .dispatch import Kuu
 
 
-@Kuu.register(InducingPoints, Mok)
-def _Kuu(feature: InducingPoints, kernel: Mok, *, jitter=0.0):
+@Kuu.register(InducingPoints, MultioutputKernel)
+def _Kuu(feature: InducingPoints, kernel: MultioutputKernel, *, jitter=0.0):
     Kmm = kernel(feature.Z, full=True, full_output_cov=True)  # [M, P, M, P]
     M = Kmm.shape[0] * Kmm.shape[1]
     jittermat = jitter * tf.reshape(tf.eye(M, dtype=Kmm.dtype), Kmm.shape)
     return Kmm + jittermat
 
 
-@Kuu.register(SharedIndependentMof, SharedIndependentMok)
-def _Kuu(feature: SharedIndependentMof,
-         kernel: SharedIndependentMok,
+@Kuu.register(FallbackSharedIndependentInducingVariables, SharedIndependent)
+def _Kuu(feature: FallbackSharedIndependentInducingVariables,
+         kernel: SharedIndependent,
          *,
          jitter=0.0):
-    Kmm = Kuu(feature.feature, kernel.kernel)  # [M, M]
+    Kmm = Kuu(feature.inducing_variable, kernel.kernel)  # [M, M]
     jittermat = tf.eye(len(feature), dtype=Kmm.dtype) * jitter
     return Kmm + jittermat
 
 
-@Kuu.register(SharedIndependentMof, (SeparateIndependentMok, SeparateMixedMok))
-def _Kuu(feature: SharedIndependentMof,
-         kernel: Union[SeparateIndependentMok, SeparateMixedMok],
+@Kuu.register(FallbackSharedIndependentInducingVariables, (SeparateIndependent, LinearCoregionalisation))
+def _Kuu(feature: FallbackSharedIndependentInducingVariables,
+         kernel: Union[SeparateIndependent, LinearCoregionalisation],
          *,
          jitter=0.0):
-    Kmm = tf.stack([Kuu(feature.feature, k) for k in kernel.kernels],
+    Kmm = tf.stack([Kuu(feature.inducing_variable, k) for k in kernel.kernels],
                    axis=0)  # [L, M, M]
     jittermat = tf.eye(len(feature), dtype=Kmm.dtype)[None, :, :] * jitter
     return Kmm + jittermat
 
 
-@Kuu.register(SeparateIndependentMof, SharedIndependentMok)
-def _Kuu(feature: SeparateIndependentMof,
-         kernel: SharedIndependentMok,
+@Kuu.register(FallbackSeparateIndependentInducingVariables, SharedIndependent)
+def _Kuu(feature: FallbackSeparateIndependentInducingVariables,
+         kernel: SharedIndependent,
          *,
          jitter=0.0):
-    Kmm = tf.stack([Kuu(f, kernel.kernel) for f in feature.features],
+    Kmm = tf.stack([Kuu(f, kernel.kernel) for f in feature.inducing_variable_list],
```

Comment:
Should `kernel.kernel` (and `kernel.kernels`) be tidied up likewise ?

Reply:
I'll leave this for later.